hw/ip/otbn/util/shared/decode.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import struct
+from typing import Dict, Tuple
+
+from shared.elf import read_elf
+from shared.insn_yaml import Insn, load_insns_yaml
+
+# Load the insns.yml file at module load time.
+try:
+    INSNS_FILE = load_insns_yaml()
+except RuntimeError as err:
+    sys.stderr.write('{}\n'.format(err))
+    sys.exit(1)
+
+
+class OTBNProgram:
+    def __init__(self, symbols: Dict[str, int], insns: Dict[int, int],
+                 data: Dict[int, int]):
+        self.symbols = symbols  # label -> PC
+        self.data = data  # addr -> data (32b word)
+
+        self.insns = {}
+        for pc in insns.keys():
+            opcode = insns[pc]
```

Comment:
Nit: `for pc, opcode in insns.items():` is probably simpler (and avoids a double lookup, not that it really matters)